src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -179,12 +179,19 @@ private BoundExpression VisitExpressionImpl(BoundExpression node)
             // like compound assignment does (extra flag only passed when it is an expression
             // statement means that this constraint is not violated).
             // Dynamic type will be erased in emit phase. It is considered equivalent to Object in lowered bound trees.
+            // Unused deconstructions are lowered to produce a return value that isn't a tuple type.
             Debug.Assert(visited == null || visited.HasErrors || ReferenceEquals(visited.Type, node.Type) ||
-                    visited.Type.Equals(node.Type, TypeCompareKind.IgnoreDynamicAndTupleNames));
+                    visited.Type.Equals(node.Type, TypeCompareKind.IgnoreDynamicAndTupleNames) ||
+                    IsUnusedDeconstruction(node));
 
             return visited;
         }
 
+        private static bool IsUnusedDeconstruction(BoundExpression node)
```

Comment:
>IsUnusedDeconstruction [](start = 28, length = 22)

Would it make sens eto put this helper as an extension method into src\Compilers\CSharp\Portable\BoundTree\BoundExpressionExtensions.cs ?